File: VB/CustomWizardExample/Wizard/DataSourceWizardCustomizationService.vb
Imports CustomWizardExample.Wizard.Presenters
Imports CustomWizardExample.Wizard.Views
Imports DevExpress.DataAccess.Wizard.Model
Imports DevExpress.DataAccess.Wizard.Views
Imports DevExpress.Snap.Services

Namespace CustomWizardExample.Wizard
    Public Class DataSourceWizardCustomizationService
        Implements IDataSourceWizardCustomizationService

        Public Sub CustomizeDataSourceWizard(ByVal tool As DevExpress.DataAccess.UI.Wizard.IWizardCustomization(Of DataSourceModel)) Implements IDataSourceWizardCustomizationService.CustomizeDataSourceWizard
            tool.WizardTitle = "Custom Data Source Wizard"

            tool.StartPage = GetType(CustomChooseDataSourceNamePage)
            tool.RegisterPage(Of CustomChooseDataSourceNamePage, CustomChooseDataSourceNamePage)()
            tool.RegisterPage(Of CustomChooseDataSourceTypePage, CustomChooseDataSourceTypePage)()
            tool.RegisterPage(Of ChoosePersonsPage, ChoosePersonsPage)()
            tool.RegisterPageView(Of IChooseDataSourceTypePageView, CustomChooseDataSourceTypePageView)()
            tool.RegisterPageView(Of IChoosePersonsPageView, ChoosePersonsPageView)()
        End Sub
    End Class
    Public Class CustomDataSourceType
        Inherits DataSourceType

        Public Shared ReadOnly XmlPersons As DataSourceType = New CustomDataSourceType()
    End Class
End Namespace

File: CS/CustomWizardExample/Wizard/DataSourceWizardCustomizationService.cs
using CustomWizardExample.Wizard.Presenters;
using CustomWizardExample.Wizard.Views;
using DevExpress.DataAccess.Wizard.Model;
using DevExpress.DataAccess.Wizard.Views;
using DevExpress.Snap.Services;

namespace CustomWizardExample.Wizard {
    public class DataSourceWizardCustomizationService : IDataSourceWizardCustomizationService {
        public void CustomizeDataSourceWizard(DevExpress.DataAccess.UI.Wizard.IWizardCustomization<DevExpress.DataAccess.Wizard.Model.DataSourceModel> tool) {
            tool.WizardTitle = "Custom Data Source Wizard";

            tool.StartPage = typeof(CustomChooseDataSourceNamePage);
            tool.RegisterPage<CustomChooseDataSourceNamePage, CustomChooseDataSourceNamePage>();
            tool.RegisterPage<CustomChooseDataSourceTypePage, CustomChooseDataSourceTypePage>();
            tool.RegisterPage<ChoosePersonsPage, ChoosePersonsPage>();
            tool.RegisterPageView<IChooseDataSourceTypePageView, CustomChooseDataSourceTypePageView>();
            tool.RegisterPageView<IChoosePersonsPageView, ChoosePersonsPageView>();
        }
    }
    public class CustomDataSourceType : DataSourceType {
        public static readonly DataSourceType XmlPersons = new CustomDataSourceType();
    }
}
